inference-engine/src/auto_plugin/auto_exec_network.cpp

```diff
@@ -133,4 +252,43 @@ Parameter AutoExecutableNetwork::GetConfig(const std::string& name) const {
     return _cacheConfig;
 }
 
+void AutoExecutableNetwork::ScheduleToWorkerInferRequest(Task inferPipelineTask, DeviceName preferred_device) {
+    // printf("%s:%d\n", __FUNCTION__, __LINE__);
+    WorkerInferRequest *workerRequestPtr = nullptr;
+    if (!preferred_device.empty()) {
+        // _acceleratorDeviceName could be the same as _cpuDeviceName, such as AUTO:CPU
+        auto &idleWorkerRequests = _idleWorkerRequests[preferred_device];
+        if (idleWorkerRequests.try_pop(workerRequestPtr)) {
+            PluginHelper::IdleGuard idleGuard{workerRequestPtr, idleWorkerRequests};
+            _thisWorkerInferRequest = workerRequestPtr;
+            {
+                auto capturedTask = std::move(inferPipelineTask);
+                capturedTask();
+            }
+            idleGuard.Release();
+            return;
+        }
+        _inferPipelineTasksDeviceSpecific[preferred_device]->push(std::move(inferPipelineTask));
+    } else {
+        // _acceleratorDeviceName could be the same as _cpuDeviceName, such as AUTO:CPU
+        auto &idleWorkerRequests = _alreadyActualNetwork
+                                       ? _idleWorkerRequests[_acceleratorDeviceName]
+                                       : _idleWorkerRequests[_cpuDeviceName];
+        if (idleWorkerRequests.try_pop(workerRequestPtr)) {
```

Comment:
this code seems to be the same as under the if(!preferred_device.empty()) on the line 257, only the idelWorkerRequests differs, it is possible to share the code with try_pop(), release quard, etc, keeping only the idleWorkers selection under the if/else

Reply:
agreed, will modify